### PR TITLE
QP use new kinsol

### PR DESCRIPTION
**Pull request by hongkai-dai**
@tkoolen , @rdeits , could you review the code?

### CODE REVIEW

**Comment by rdeits** [comment]
I don't see any problems, but I'm also not familiar enough with the new/old kinematics code to be all that helpful here

File: systems/controllers/QPCommon.cpp [hunk 1]
```diff
@@ -534,26 +551,28 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
...
-  MatrixXd Jbdot(6,nq);
+  Vector6d Jbdotv;	
   for (int i=0; i<desired_body_accelerations.size(); i++) {
     if (desired_body_accelerations[i].weight < 0) { // negative implies constraint
       if (!inSupport(active_supports,desired_body_accelerations[i].body_id0)) {
-        pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig,1,Jb);
-        pdata->r->forwardJacDot(desired_body_accelerations[i].body_id0,orig,1,Jbdot);
+        pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig1,1,Jb);
+        auto Jdot_times_v = pdata->r->forwardJacDotTimesV(orig,desired_body_accelerations[i].body_id0,0,1,0);
+        Jbdotv = Jdot_times_v.value();
```
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: For future reference, this performs an unnecessary copy from `Jdot_times_v.value()` to `Jbdotv`. You can get rid of the copy by making Jbdotv a reference to the appropriate type (either auto& or I suppose Vector6d&). Not essential to fix this right now but just so you know.

File: systems/controllers/QPCommon.cpp [hunk 2]
```diff
@@ -706,13 +726,14 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
...
         if (!inSupport(active_supports,desired_body_accelerations[i].body_id0)) {
-          pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig,1,Jb);
-          pdata->r->forwardJacDot(desired_body_accelerations[i].body_id0,orig,1,Jbdot);
+          pdata->r->forwardJac(desired_body_accelerations[i].body_id0,orig1,1,Jb);
+          auto Jdot_times_v = pdata->r->forwardJacDotTimesV(orig,desired_body_accelerations[i].body_id0,0,1,0);
+          Jbdotv = Jdot_times_v.value();
 
           for (int j=0; j<6; j++) {
             if (!std::isnan(desired_body_accelerations[i].body_vdot[j])) {
               pdata->Hqp += desired_body_accelerations[i].weight*(Jb.row(j)).transpose()*Jb.row(j);
```
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: You could save a copy into a temporary here by using .noalias() on the left hand side.

File: systems/controllers/QPCommon.cpp [hunk 3]
```diff
@@ -706,13 +726,14 @@ int setupAndSolveQP(NewQPControllerData *pdata, std::shared_ptr<drake::lcmt_qp_c
...
-              f.head(nq) += desired_body_accelerations[i].weight*(robot_state.qd.transpose()*Jbdot.row(j).transpose() - desired_body_accelerations[i].body_vdot[j])*Jb.row(j).transpose();
+              f.head(nq) += desired_body_accelerations[i].weight*(Jbdotv(j) - desired_body_accelerations[i].body_vdot[j])*Jb.row(j).transpose();
```
Comment (tkoolen) [inline systems/controllers/QPCommon.cpp]: You could save a copy into a temporary here by using .noalias() on the left hand side.

File: systems/controllers/QPControllermex.cpp [hunk 4]
```diff
@@ -304,23 +302,27 @@ void mexFunction(int nlhs, mxArray *plhs[], int nrhs, const mxArray *prhs[])
   
   pdata->r->getCOM(xcom);
   pdata->r->getCOMJac(pdata->J);
-  pdata->r->getCOMJacDot(pdata->Jdot);
+  MatrixXd Jdot;
+  pdata->r->getCOMJacDot(Jdot);
+  pdata->Jdotv = Jdot*qd;
   pdata->J_xy = pdata->J.topRows(2);
-  pdata->Jdot_xy = pdata->Jdot.topRows(2);
+  pdata->Jdotv_xy = pdata->Jdotv.head(2);
```
Comment (tkoolen) [inline systems/controllers/QPControllermex.cpp]: for these, using the fixed size version could be alittle faster (e.g. `Jdotv.head<2>()`)

**Comment by kuindersma** [comment]
Did you run a quick profile on walking to make sure there isn't a performance difference?

**Comment by hongkai-dai** [comment]
@kuindersma , good call. I tested it in runAtlasWalkingSplit. 
On this branch,
1. with use_new_kinsol = true, the simulation takes 133 seconds
2. with use_new_kinsol = false, the simulation takes 129 seconds.
On the master branch (which uses the old kinsol), it takes 130 seconds. 
I think the running time can be regarded as unchanged.

**Comment by kuindersma** [comment]
Great, thanks for doing that!

**Comment by hongkai-dai** [comment]
I think this one is ready to be merged?

**Pull request merged**